bgpd/bgp_bmp.c

```diff
@@ -2268,10 +2268,12 @@ DEFPY(bmp_monitor_cfg,
 			bmp->afistate[afi][safi] = BMP_AFI_INACTIVE;
 			continue;
 		}
-
 		bmp->afistate[afi][safi] = BMP_AFI_NEEDSYNC;
 	}
 
+	frr_each(bmp_session, &bt->sessions, bmp)
```

Comment:
with several FULL bgp sessions up and running won't this cause event issues?  Would it be better to schedule an event that happens in the future to do all this work?  Or to break up the calls per session?

Reply:
agreed ... we shouldn't walk the entire list of sessions on an event like this ... it could well cause problems in operation

Reply:
without that patch, current BMP behavior is : 
- after BMP reconfiguration, when receiving new incoming BGP update, the whole routing table is parsed again.

some additional optimisations will be needed.